File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math
def solution(progresses, speeds):
    answer = []
    days = [0] * len(progresses)
    cnt = 1
    
    for i in range(len(progresses)):
        days[i] = math.ceil((100-progresses[i]) / speeds[i])
    
    standard = days[0]
    for i in range(1, len(days)):
        if days[i] <= standard:
            cnt += 1
        else:
            answer.append(cnt)
            standard = days[i]
            cnt = 1
    
    answer.append(cnt)
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    static class Work {
        int idx;
        int duration;
        
        Work(int idx, int duration) {
            this.idx = idx;
            this.duration = duration;
        }
    }
    
    public int[] solution(int[] progresses, int[] speeds) {
        Work[] works = new Work[speeds.length];
        
        for (int i = 0; i < progresses.length; i++) {
            works[i] = new Work(i, (int) Math.ceil((100 - progresses[i]) / (double) speeds[i]));
        }
        
        ArrayList<Integer> answer = new ArrayList<>();
    
        
        int cnt = 1;
        int curDuration = works[0].duration;
        
        for (int i = 1; i < works.length; i++) {
            if (works[i].duration > curDuration) {
                answer.add(cnt);
                cnt = 1;
                curDuration = works[i].duration;
                continue;
            }
            
            cnt++;
        }
    
        answer.add(cnt);
        
        return answer.stream()
            .mapToInt(Integer::intValue)
            .toArray();
    }
}